modules/microshift-custom-ca-proc.adoc

```diff
@@ -0,0 +1,221 @@
+// Module included in the following assemblies:
+//
+// * microshift_security_compliance/microshift-custom-ca.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-custom-cas-configuring_{context}"]
+= Configuring custom certificate authorities
+
+To configure externally generated certificates and domain names using custom certificate authorities (CAs), add them to the {microshift-short} `/etc/microshift/config.yaml` configuration file. You must also configure the host operating system trust root.
+
+[NOTE]
+====
+Externally generated `kubeconfig` files are created in the `/var/lib/microshift/resources/kubeadmin/<hostname>/kubeconfig` directory. If you need to use `localhost` in addition to externally generated configurations, retain the original `kubeconfig` file in its default location. The `localhost` `kubeconfig` uses the self-signed certificate authority.
+====
+
+.Prerequisites
+* The OpenShift CLI (`oc`) is installed.
+* You have access to the cluster as a user with the cluster administration role.
+* Certificate authorities are set up.
+* A {microshift-short} `/etc/microshift/config.yaml` configuration file exists.
+
+[IMPORTANT]
+====
+Either replace the `certificate-authority-data` in the external `kubeconfig` with the new `rootCA`, or add the parameter to the trust root of the operating system. Do not do both.
+====
+
+.Procedure
+
+. For each custom CA that you need, add an `apiServer` section called `namedCertificates` to the `/etc/microshift/config.yaml` {microshift-short} configuration file by using the following example:
++
+[source,yaml]
+----
+apiServer:
+  namedCertificates:
+   - certPath: ~/certs/api_fqdn_1.crt <1>
+     keyPath:  ~/certs/api_fqdn_1.key <2>
+   - certPath: ~/certs/api_fqdn_2.crt
+     keyPath:  ~/certs/api_fqdn_2.key
+     names: <3>
+     - api_fqdn_1
+     - *.apps.external.com
+----
+<1> Add the full path to the certificate.
+<2> Add the full path to the certificate key.
+<3> Optional. Add a list of explicit DNS names. Leading wildcards are allowed. If no names are provided, the implicit names are extracted from the certificates.
+
+. Restart the {microshift-service} to apply the CAs by running the following command:
++
+[source,terminal]
+----
+$ systemctl microshift restart
+----
+
+. Wait five minutes for system restart and apply the custom server.
+
+. Check that the certificates are applied by using the following command:
++
+[source,terminal]
+----
+$ oc --insecure-skip-tls-verify get node
```

Comment:
1. before using the `oc` command we need to specify the kubeconfig `export KUBECONFIG=/var/lib/microshift/resources/kubeadmin/<custom_cert_fqdn>/kubeconfig`
2. instead of using insecure we could explicitly specify the CA.
`oc --certificate-authority ~/certs/ca.ca get node`

Reply:
Chuckling at a file called `ca.ca`.